ProjectLighthouse.Servers.Website/Pages/UserSettingsPage.cshtml.cs

```diff
@@ -41,12 +41,21 @@ public UserSettingsPage(DatabaseContext database) : base(database)
 
         string? avatarHash = await FileHelper.ParseBase64Image(avatar);
 
-        if (avatarHash != null) this.ProfileUser.IconHash = avatarHash;
+        if (avatarHash != null)
+        {
+            // Deny request if in read-only mode
+            if (ServerConfiguration.Instance.UserGeneratedContentLimits.ReadOnlyMode) return this.BadRequest();
```

Comment:
```suggestion
            if (ServerConfiguration.Instance.UserGeneratedContentLimits.ReadOnlyMode) return this.Redirect($"~/users/{userId}");
```
This should probably redirect back to the user's page rather than returning a bad request so that they don't get stuck on the 400 page.